BCP195bis/draft-ietf-uta-rfc7525bis.md

```diff
@@ -601,6 +601,15 @@ random, partially implicit nonces, i.e., the nonce construction used in TLS
 it is RECOMMENDED that the same limit of 2<sup>24.5</sup> records is used for both
 versions.
 
+For DTLS 1.2 where, unlike TLS, the established security context is not torn
+down on the first decryption failure, there is a separate bound for the number
+of forgeries that can be tolerated before the current session keys become
+insecure.  The formulas in Section 6.1 of {{?I-D.irtf-cfrg-aead-limits}}
+related to random, partially implicit nonces, provide an upper bound for the
+number of forgery attempts (i.e., failed AEAD decryption invocations) of
+2<sup>28</sup>.  Note that since this value is higher than the confidentiality
+bound recommended above, the latter can be used instead.
+
```

Comment:
This is very confusing (for me and probably for developers). What is the maximum number of records I am allowed to *send*? As a receiver, am I really expected to tolerate millions of incorrect decryptions per connection? I understand why you should tolerate a few (because of random packet corruption and to not be too susceptible to DoS) but this sounds like too much.

Reply:
> This is very confusing (for me and probably for developers). What is the maximum number of records I am allowed to _send_?

(On each direction) sender and receiver share the same key, so they share the same confidentiality bound.  Whoever gets to the threshold first will kill the session and resume/re-handshake.  The integrity bound instead makes sense on the receiver side only.

> As a receiver, am I really expected to tolerate millions of incorrect decryptions per connection? I understand why you should tolerate a few (because of random packet corruption and to not be too susceptible to DoS) but this sounds like too much.

The situation today is that most DTLS stacks silently drop a packet that failed decryption on the floor, so effectively they tolerate an infinite number of corruptions.  In fact, this is more or less what the [spec tells us to do](https://datatracker.ietf.org/doc/html/rfc6347#section-4.1.2.1).